edi/models/edi_exchange_record.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2020 ACSONE SA
+# @author Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, exceptions, fields, models
+
+
+class EDIExchangeRecord(models.Model):
+    """
+    Define an exchange record.
+    """
+
+    _name = "edi.exchange.record"
+    _inherit = "mail.thread"
+    _description = "EDI exchange Record"
+    _order = "exchanged_on desc"
+
+    name = fields.Char(compute="_compute_name")
+    type_id = fields.Many2one(
+        string="EDI Exchange type",
+        comodel_name="edi.exchange.type",
+        required=True,
+        ondelete="cascade",
+        auto_join=True,
+    )
+    direction = fields.Selection(related="type_id.direction")
+    backend_id = fields.Many2one(
+        comodel_name="edi.backend", related="type_id.backend_id"
+    )
+    model = fields.Char(index=True, required=False, readonly=True)
+    res_id = fields.Many2oneReference(
+        string="Record ID",
+        index=True,
+        required=False,
+        readonly=True,
+        model_field="model",
+    )
+    exchange_file = fields.Binary(attachment=True)
+    exchange_filename = fields.Char(
+        compute="_compute_exchange_filename", readonly=False, store=True
+    )
+    exchanged_on = fields.Datetime(
+        string="Exchanged on",
+        help="Sent or received on this date.",
+        compute="_compute_exchanged_on",
+        store=True,
+    )
+    # TODO: use sequence and make it unique
+    exchange_identification_code = fields.Char(
+        track_visibility="onchange",
+        help="Identification of the EDI, useful to search and join other documents",
+    )
+    ack_file = fields.Binary(attachment=True)
```

Comment:
@simahawk just a question here, wouldn't this be more suitable in a new exchange record type "ack"?